src/main/java/org/hisp/dhis/rules/models/RuleVariable.java

```diff
@@ -21,8 +18,29 @@
     @Nonnull
     public abstract String name();
 
+    public abstract boolean useCodeForOptionSet();
+
+    @Nonnull
+    public abstract List<Option> options();
+
     public abstract Map<String, RuleVariableValue> createValues( RuleVariableValueMapBuilder builder,
         Map<String, List<RuleDataValue>> allEventValues,
         Map<String, RuleAttributeValue> currentEnrollmentValues,
         Map<String, RuleDataValue> currentEventValues );
+
+    public String getOptionName( String value )
+    {
+        // if no option found then existing value in the context will be used
+        String optionName = value;
+
+        for ( Option op : options() )
+        {
+            if (op.getCode().equals( value ) )
+            {
+                optionName = op.getName();
+            }
+        }
+
+        return optionName;
+    }
```

Comment:
```suggestion
    public String getOptionName( String value )
    {
        if (options() == null || options().isEmpty()) {
            return value;
        }

        return options().stream()
                .filter(op -> Objects.equals(value, op.getCode()))
                .map(Option::getName)
                .findAny()
                .orElse(value);
    }
```

Having a guard clause make it easier to understand that if there are no options the value should be used instead.
I am still not sure if the option is not there if we should use the value then or if we should raise an exception